### PR TITLE
Added favicon

### PR DESCRIPTION
**Pull request by Yakimych**

### CODE REVIEW

**Comment by Yakimych** [comment]
Note that when using the favicons-webpack-plugin, quite a lot of icons get generated. Most (probably all) of them are also included into index.html, since I turned on the `inject` option in html-webpack-plugin too.

**Comment by MargaretKrutikova** [comment]
Wow, that looks awesome! Thanks a lot for the icon! ❤️ 
I will just click around a bit and see whether nothing broke and merge it.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ npm-debug.log
 *.bs.js
 .graphql_ppx_cache
 coverage
+build
```
Comment (MargaretKrutikova) [inline .gitignore]: How could this have been missing for so long ... 😄

File: package.json [hunk 2]
```diff
@@ -37,7 +37,8 @@
     "@glennsl/bs-jest": "^0.4.9",
     "bs-platform": "^5.0.6",
     "css-loader": "^3.2.0",
-    "html-webpack-plugin": "^3.2.0",
+    "favicons-webpack-plugin": "^3.0.1",
+    "html-webpack-plugin": "^4.0.0-beta.11",
```
Comment (MargaretKrutikova) [inline package.json]: Why is this one beta now?

File: webpack.config.js [hunk 3]
```diff
@@ -1,20 +1,30 @@
...
     path: outputDir,
+    publicPath: "/",
     filename: "Index.js"
   },
   plugins: [
     new HtmlWebpackPlugin({
       template: "src/index.html",
-      inject: false
+      favicon: "src/favicon.ico",
+      inject: true
```
Comment (MargaretKrutikova) [inline webpack.config.js]: Why would you ever want `inject` to be `false`, like it was before?

File: package.json (hunk 2 above)
Comment (Yakimych) [reply package.json]: Not sure, but the `favicons-webpack-plugin` wants this version of `html-webpack-plugin`.

File: webpack.config.js (hunk 3 above)
Comment (Yakimych) [reply webpack.config.js]: Good question - my webpack knowledge is rather limited, but my guess is that it's fine for simple cases like the default bucklescript template, where it's enough to include some static file.

**Comment by MargaretKrutikova** [comment]
Sorry, merged my branch a bit too fast and now you have conflicts. Could you resolve them or shall I do it?

**Comment by MargaretKrutikova** [comment]
Great job! Merged the new icon 😎

**Pull request merged**